### ISSUE TITLE
Master branch is writing everything to a single index

### ISSUE THREAD

**Issue by axw** [elastic/apm-server#1610]
This is potentially specific to how apm-integration-testing configures the server, but I started a stack with

    apm-integration-testing/scripts/compose.py start --with-opbeans-go master

and all the APM data is going into a single index, ignoring the "output.elasticsearch.indices" config in apm-server.yml.

```
$ curl http://localhost:9200/_cat/indices
green open .monitoring-es-6-2018.11.30     35NLZbU4SBqurzUXNj8CJw 1 0  2660 56   1.7mb   1.7mb
green open .kibana_1                       TFg_KjziRvaA1mDpS8xclQ 1 0    19  0  55.8kb  55.8kb
green open .watches                        0rxzXk-uRQeJmzsiibwmtg 1 0     6  0  26.1kb  26.1kb
green open .monitoring-beats-6-2018.11.30  0RPiuvViTY6EtBRzW0ePVg 1 0   233  0 341.7kb 341.7kb
green open apm-7.0.0-2018.11.30            wpVI444rRtejX7u9jVlG8A 1 0 25307  0   4.8mb   4.8mb
green open .triggered_watches              QSp9M2jpQo2euVtpMnsrRA 1 0     0  0    230b    230b
green open .watcher-history-9-2018.11.30   OKjpIjqdScuP3akT-qNK2w 1 0   198  0 487.8kb 487.8kb
green open .monitoring-kibana-6-2018.11.30 cadXu4QuSluI1iHKmK4ftA 1 0   202  0 119.3kb 119.3kb
```

**Comment by graphaelli** [comment]
My first thought was that this is resulting from the docker images changes that just landed, but I've confirmed the same behavior even after ensuring the index splitting configuration is in place:
```
output.elasticsearch:
  indices:
    - index: "apm-%{[beat.version]}-error-%{+yyyy.MM.dd}"
      when.contains:
        processor.event: "error"
...
```
and the bundled docker image configuration is also correct.  I've also confirmed 6.x still works as expected.  I'll continue investigating tomorrow if no one else has picked it up by then.

**Comment by graphaelli** [comment]
tracked this down to #1534 where libbeat moved `beat.version` to `agent.version`.  Make the corresponding change in the `index` directives does the expected.   I want `agent.version` in APM to be the version of the agent reporting the data, and use something else for the apm-server version, currently targeting `agent.server.version`, but that doesn't exist yet.

**Comment by simitt** [comment]
As discussed offline I looked into that, and can confirm your first suspicion that changing `beat.version` to `agent.version` was the root cause (sorry for the confusion).

Meanwhile a lot of ECS related changes went in. On the APM Server side the [`observer` top level field has been added](https://github.com/elastic/apm-server/blob/master/publish/pub.go#L72), which replaces basically the former `beat` information. We can make use of this to change from `beat.version` to `observer.version` in the config file, and the problem for the index is solved. 

However, we also need to find a solution for other parts that use `observer` in the config file, such as `templates`. For `templates` currently [only `beat` information]() is replaced when creating the template, which leads to an error when using something else than `beat` in the name. 

@ruflin as the change from `host` to `agent` was done with [this PR change](https://github.com/elastic/beats/commit/1767034c0527752fc35cb2e1b60fe708e860e8d1#diff-73bd591578392eb2d281f4778ac9a47e), have you thought about which field we should use for other places and how to replace `beat` in the config files? For us it is not proper to use `agent`, as the agent is reflecting other information.
